### PR TITLE
Bug 1975475: aws: block creation of bootstrap instance until ignition config is uploaded

### PR DESCRIPTION
**Pull request by staebler**
Give the bootstrap instance a dependency on the creation of the s3 bucket object holding the bootstrap's ignition config. The bootstrap instance requires the ignition config in order to complete its creation. It takes around 2 minutes to create the bucket object, and the instance creation has a 2-minute timeout. Consequently, there are times when the bootstrap fails to complete in time because the creation of the bucket object took too long.

### CODE REVIEW

**Comment by staebler** [comment]
This PR is 4 for 4 on successful AWS installs, so I think it is good.
/cc @jhixson74

**Comment by staebler** [comment]
These changes do not affect upgrades.
/override e2e-aws-upgrade

**Review by e-tienne — COMMENTED** [review]
/lgtm

**Comment by staebler** [comment]
> These changes do not affect upgrades.
> /override e2e-aws-upgrade

Let me try again with the correct syntax.
/override ci/prow/e2e-aws-upgrade
